protostar/commands/test/starkware/forkable_starknet.py

```diff
@@ -15,6 +16,13 @@ class CheatableCarriedState(CarriedState):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.pranked_contracts_map: Dict[int, int] = {}
+        self.event_selector_to_name_map: Dict[int, str] = {}
+
+    def update_event_selector_to_name_map(
```

Comment:
Why do we copy this data? Can't we just provide property to ForkableStarknet which maps to this map? This way we would probably be compatible with new deploy syscall

Reply:
I can move this logic to `ForkableStarknet` if this is what you mean. `event_selector_to_name_map` stores mapping for all contracts whereas `local_event_selector_to_name_map` only for a single contract, that's why I need to copy the map.

Why do we need `self.pranked_contracts_map` and `CheatableCarriedState` then?

Reply:
Ok, we will have to investigate it with cairo 0.9 but for now write a test and we can merge this